addon/controller.js

```diff
@@ -10,16 +15,24 @@ var defaultCodes = {
 
 export default {
 
-  handle: function(verb, handler, db, request, code) {
-    code = code ? code : (defaultCodes[verb] || 200);
-
+  handle: function(verb, handler, db, request, customizedCode) {
+    var code, isEmptyObject;
     var handlerMethod = this._lookupHandlerMethod(verb, handler);
-    var response = handlerMethod(handler, db, request, code);
+    var response = handlerMethod(handler, db, request);
 
     if (response instanceof Response) {
       return response.toArray();
 
     } else {
+      if (customizedCode) {
+        code = customizedCode;
```

Comment:
If a status code is specified by the user, it takes precedence over everything else.